### DISCUSSION TITLE
Manage multi schemas on postgres

### DISCUSSION THREAD

**Discussion by Rowmain** [paratestphp/paratest#818] [Q&A]
When using multi schemas on postgres, only the default one is handle by the "paratest".
The others one, furthermore if they have foreign key, do not have their database connection configuration changed.

**Comment by Slamdunk** [comment]
Hi, I have no idea what you're talking about as I don't use PostgreSQL; btw it's up to you to redirect each subprocess to use the a different schema, see for example https://github.com/paratestphp/paratest#test-token

**Comment by mfn** [comment]
I've a feeling this is related to the support of paratest in Laravel via https://laravel.com/docs/8.x/testing#running-tests-in-parallel

I know that this implementation only works with the default "database connection" (though OP is talking about schemas, so who knows).

**Comment by robsontenorio** [comment]
@Rowmain @Slamdunk  @mfn 

Do you guys have found any solution for Laravel?

The problem with Postgres is when you use a schema name different than `public`.



- Works : php artisan test
- Works: php artisan test --parallel

```
'pgsql' => [
            'driver' => 'pgsql',
            ....
            'schema' => 'public',
        ],
```


- Works : php artisan test
- **Dont Work**: php artisan test --parallel

```
'pgsql' => [
            'driver' => 'pgsql',
            ....
            'schema' => 'myblog',
        ],
```

Paratest always tries to use `public` schema.

This is a problem, because sometimes you don't want to use `public` schema for security reasons.

**Comment by Costin-Nox** [comment]
I have a project that uses multiple schemas in laravel, can confirm behaviour above is still the same, works if you don't use --parallel 

I'm guessing there's no workaround for this?
